### DISCUSSION TITLE
Unable to reset authentication on Azure container instance

### DISCUSSION THREAD

**Discussion by derekgrus** [datalust/seq-tickets#1963] [Help]
Hello,

 Our app secret has expired. I followed the steps [here](https://docs.datalust.co/docs/recovering-from-broken-authentication-settings#resetting-the-authentication-provider-and-unlocking-the-user-account)

but i am getting this error.
![2023-09-20 07_54_37-ci-seq-prod-westus3-001 - Microsoft Azure and 6 more pages - Work - Microsoft​ E](https://github.com/datalust/seq-tickets/assets/89416010/627885e0-db2f-4a92-a6ed-846426e860aa)

I tried to run flare and got this.

![2023-09-20 07_56_32-ci-seq-prod-westus3-001 - Microsoft Azure and 6 more pages - Work - Microsoft​ E](https://github.com/datalust/seq-tickets/assets/89416010/3f142336-2a29-488c-a70b-1a5d4ffabdb3)

**Comment by KodrAus** [comment]
Hi @derekgrus 👋 The problem here is that those docs assume you’re running on Windows and can stop and start the service at will, but Seq is always running in a Docker container so you can’t stop the process or make changes to its config from within the container itself.

What you can do is stop that container through docker, and then start a fresh temporary one to fix up the auth config. We’ve got some docs on that [here](https://docs.datalust.co/docs/getting-a-shell-to-a-container#using-init-scripts).

Hope that helps!

EDIT: Checking those docs we do have steps for Docker. Is that error what you’re seeing in a temporary container after stopping your main one?

**Reply by derekgrus** [reply]
@KodrAus I am running this in a linux container.

**Reply by KodrAus** [reply]
Is this after `docker stop`ing your Seq container and running the command in a temporary one?

**Reply by derekgrus** [reply]
the docker command does not work.
![2023-09-20 08_55_43-ci-seq-prod-westus3-001 - Microsoft Azure and 8 more pages - Work - Microsoft​ E](https://github.com/datalust/seq-tickets/assets/89416010/7f59fcca-874e-4046-9daf-e2219f7a80d3)

**Reply by KodrAus** [reply]
It looks like that shell might actually be in your Docker container? You’ll need to run the command on the Docker host rather than from the container. If you’re in a cloud environment where you can’t access that host directly you should be able to use its UI to stop your container.

**Reply by derekgrus** [reply]
i did stop the container,  but don't have console access

**Reply by KodrAus** [reply]
I’m not quite sure what environment you’re hosting your container in, but if you’re not able to stop your container and get a shell to a new one then you can also put the reset commands in [an init script](https://docs.datalust.co/docs/init-scripts) and then restart your container.

**Reply by derekgrus** [reply]
it is hosted in an azure container instance
![2023-09-20 09_25_17-ci-seq-prod-westus3-001 - Microsoft Azure and 10 more pages - Work - Microsoft​ ](https://github.com/datalust/seq-tickets/assets/89416010/52214c32-18f7-4152-b72f-10c20ed78518)

**Reply by derekgrus** [reply]
i uploaded the script, but now my nginx container does not start.

![2023-09-20 09_33_12-seq-data - Microsoft Azure and 13 more pages - Work - Micr
![2023-09-20 09_35_52-ci-seq-prod-westus3-001 - Microsoft Azure and 13 more pages - Work - Microsoft​ ](https://github.com/datalust/seq-tickets/assets/89416010/7dc213fd-8cbc-4e79-9304-a7c98f32c872)
osoft​ Edge](https://github.com/datalust/seq-tickets/assets/89416010/16075344-9c6d-4529-995b-327b92860afc)



![2023-09-20 09_35_52-ci-seq-prod-westus3-001 - Microsoft Azure and 13 more pages - Work - Microsoft​ ](https://github.com/datalust/seq-tickets/assets/89416010/89c1166a-235a-4c1c-88f0-d237932ada78)

**Reply by nblumhardt** [reply]
Hi Derek, thanks for the update.

What's the state of the Seq container at this point? Did it successfully restart after running the script, or is it stuck in some state? (Logs from the Seq container should help.)

If the script ran, it will have caused a `.done` file to be created in the `/data/Init` directory, so this would be one thing to check. The storage screenshot you've sent doesn't show what path we're looking at - assuming that's `/seq-init` but let me know if it's a different directory.

Best regards,
Nick

**Reply by derekgrus** [reply]
I ended up removing the file to get it back online. there was no .done in the data/Init folder.

Should I email the logs or attach them here?

![2023-09-20 16_42_25-ci-seq-prod-westus3-001 - Microsoft Azure and 1 more page - Work - Microsoft​ Ed](https://github.com/datalust/seq-tickets/assets/89416010/2ce89952-aaf4-46de-a334-784f1a541eb2)

**Reply by nblumhardt** [reply]
Hi Derek - emailing us would be good, thanks 👍

**Reply by derekgrus** [reply]
Email sent! Thanks @nblumhardt

**Reply by derekgrus** [reply]
hey @nblumhardt ,

 Here are the contents of the az container logs.
 
 
```
PS /home/derek> az container logs --resource-group rg-nginx-proxy-prod-001 --name ci-seq-prod-westus3-001
2023/09/21 13:03:02 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:02 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
10.92.0.24 - - [21/Sep/2023:13:03:04 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:04 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:14 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:14 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:14 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:14 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
2023/09/21 13:03:16 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:16 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:26 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:26 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"

PS /home/derek> az container logs --resource-group rg-nginx-proxy-prod-001 --name ci-seq-prod-westus3-001
2023/09/21 13:03:02 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:02 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
10.92.0.24 - - [21/Sep/2023:13:03:04 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:04 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:14 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:14 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:14 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:14 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
2023/09/21 13:03:16 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:16 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:26 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:26 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"

PS /home/derek> az container logs --resource-group rg-nginx-proxy-prod-001 --name ci-seq-prod-westus3-001
2023/09/21 13:03:02 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:02 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
10.92.0.24 - - [21/Sep/2023:13:03:04 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:04 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:14 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:14 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:14 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:14 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
2023/09/21 13:03:16 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:16 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:26 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:26 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
10.92.0.24 - - [21/Sep/2023:13:03:34 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:34 [error] 1950#1950: *2 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:44 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:44 [error] 1952#1952: *17 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
2023/09/21 13:03:44 [error] 1950#1950: *16 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.26, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.26 - - [21/Sep/2023:13:03:44 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:46 [error] 1950#1950: *6 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:46 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
2023/09/21 13:03:46 [error] 1952#1952: *17 connect() failed (111: Connection refused) while connecting to upstream, client: 10.92.0.24, server: localhost, request: "POST /api/events/raw HTTP/1.1", upstream: "http://127.0.0.1:80/api/events/raw", host: "seqlogs.contoso.com"
10.92.0.24 - - [21/Sep/2023:13:03:46 +0000] "POST /api/events/raw HTTP/1.1" 502 157 "-" "-"
```

**Reply by nblumhardt** [reply]
Hi @derekgrus,

Unfortunately those look like the nginx logs, which will be more plentiful and recent than the ones from the `datalust/seq` container in the set.

`az container logs` has a `--container-name` argument that should get you the logs for the Seq container only.

If they don't look promising (i.e. there's something repetitive drowning out the logs from the restart), then trying to force a container restart to capture the output might help.

Getting the output from a restart attempt is critical for getting you back up and running.

Just adding a couple of other points to consider:

 * Is there still a copy of an init script in there somewhere? (A failing init script could cause the restart failure, it'd be good to get a handled on where that might be coming from.)
 * Is the `/data` mount a Premium Files share? (If not, let us know; SMB shares/non-Premium Azure Files shares suffer some problems that could be involved.)

The container logs from a restart are the critical info, though.

Hope this helps,
Nick

**Reply by misterpalmer** [reply]
Thanks @nblumhardt and team. Using an init script to show users and then a second init script to reset, I was able to reset the authentication to username/password. I was subsequently prompted to change the password on initial login, as stated in the documentation. Now, I'm successfully seeing the logs within the app. I'll complete the remainder of the full reset process to get restore AAD authentication.

Appreciate the help.
Matt
